instrumentation/java-http-server/library/src/main/java/io/opentelemetry/instrumentation/httpserver/internal/JdkHttpServerAttributesGetter.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.httpserver.internal;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpsExchange;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpServerAttributesGetter;
+import java.net.InetSocketAddress;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+enum JdkHttpServerAttributesGetter
+    implements HttpServerAttributesGetter<HttpExchange, HttpExchange> {
```

Comment:
when the request and response are the same I think we usually use `<HttpExchange, Void>`

Reply:
doesn't seem to work as the tests fail for lack of status code

Reply:
Sorry, I misremembered. As you noticed can't just use `null` for response even if it is the same as the request.